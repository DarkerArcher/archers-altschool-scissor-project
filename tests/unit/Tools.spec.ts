import {describe, it, expect} from '@jest/globals'
import { mount } from '@vue/test-utils'
import ToolsVue from '@/views/ToolsComponent.vue'

describe('Tools', () => {
    it('should render correctly', () => {
        const wrapper = mount(ToolsVue)
        expect(wrapper.html()).toMatchSnapshot()
    })

    it('should render the correct markup', () => {
        const wrapper = mount(ToolsVue)
        expect(wrapper.html()).toContain('<div class="tools-container">')
    })
})