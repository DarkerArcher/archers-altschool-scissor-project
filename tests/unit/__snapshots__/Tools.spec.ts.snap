// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tools should render correctly 1`] = `
<div class="tools-container">
  <div class="header">
    <div class="header-logo">
      <router-link to="/">
        <div class="logo-link"><img style="width: 24px; height: 24px;">
          <h2 class="logo-text"> CHOP </h2>
        </div>
      </router-link>
    </div>
    <div class="sign-in-out">
      <router-link to="/login"><button class="sign-in-button">Sign In</button></router-link>
    </div>
  </div>
  <div class="tools-content">
    <router-link to="/shortener">
      <div class="tools-items"><img style="width: 100px; height: 100px;">
        <h3 class="tool-name"> URL Shortener </h3>
      </div>
    </router-link>
    <router-link to="/qr-code">
      <div class="tools-items"><img style="width: 100px; height: 100px;">
        <h3 class="tool-name"> QR Code Generator </h3>
      </div>
    </router-link>
  </div>
</div>
`;
